### DISCUSSION TITLE
Conditionally installing a certain ComponentGroup

### DISCUSSION THREAD

**Discussion by hypetsch** [wixtoolset/issues#8598] [Questions]
We want to modify our installer to optionally exclude a specific module, for example by using a parameter like "NO_TOOL=1".

Currently the simplified structure looks like this:

```xml
<ComponentGroup Id="Tool" Directory="TOOL_DIRECTORY">
   <!-- Executable -->
   <Component>
      <File Source="$(Tool.TargetPath)" KeyPath="yes">
         ...
      </File>
   </Component>

   <!-- Dependencies -->
   <Files Include="$(Tool.TargetDir)/**">
      ...
   </Files>
</ComponentGroup>

<Feature Id="Feature1" Level="1000">
   <ComponentGroupRef Id="Tool" />
   <ComponentGroupRef Id="Module1" />
</Feature>

<Feature Id="Feature2" Level="1000">
   <ComponentGroupRef Id="Tool" />
   <ComponentGroupRef Id="Module2" />
</Feature>

<Feature Id="Feature3" Level="1000">
   <ComponentGroupRef Id="Module3" />
</Feature>
```

...where `Tool` is a utility that is, by default, installed alongside some features (`Feature1` and `Feature2`), but not with `Feature3`. Deploying Tool should remain the default and require no user interaction, but there should be an option to disable its deployment.

What would be the best way to achieve this?

**Comment by OlegSirenko** [comment]
I think the best solution for your case is usage of [File's](https://wixtoolset.org/docs/schema/wxs/file/) condition attribute:
`Condition (String) : [WiX v5] Only valid when the File element is not a child of the Component element. The condition expression to be evaluated at install time. When false the File will not be installed.`
